ext_test.go

```diff
@@ -46,8 +47,10 @@ func TestHungRequest(t *testing.T) {
 		})
 	}
 
-	require.NoError(t, hosts[0].Peerstore().AddProtocols(hosts[1].ID(), protocol.ConvertToStrings(d.serverProtocols)...))
-	d.peerFound(ctx, hosts[1].ID(), true)
+	// Wait at most 100ms for a peer in our routing table.
+	for i := 0; i < 100 && d.routingTable.Size() == 0; i++ {
+		time.Sleep(10 * time.Millisecond)
+	}
```

Comment:
This test still fails: https://github.com/libp2p/go-libp2p-kad-dht/issues/623#issuecomment-624413116. I got

```
=== RUN   TestHungRequest
    TestHungRequest: ext_test.go:69: expected to fail with deadline exceeded, got: %!s(<nil>)
    TestHungRequest: ext_test.go:73: GetClosestPeers should not have returned yet failed to find any peer
 in table
    TestHungRequest: ext_test.go:83: expected peers in the routing table
--- FAIL: TestHungRequest (1.16s)
```

Reply:
I'm pretty sure this was failing for me, but maybe I had some stale state. Will try and run the test more times and report back.

Reply:
You definitely had the change ("expected peers in the routing table" is new).

Reply:
Ah, you were probably right about the identify race.

Reply:
k, just as a heads up though it's much harder to reproduce the error locally now then previously

Reply:
Ok, I think I've fixed it now by just not connecting till everything has been setup.